src/extensions/scratch3_vizblocks/index.js

```diff
@@ -136,6 +155,197 @@ class Scratch3VizBlocks {
+                        default: 'draw line',
+                        description: 'draw line'
+                    })
+                },
+                {
+                    opcode: 'clear',
+                    blockType: BlockType.COMMAND,
+                    text: formatMessage({
+                        id: 'vizblocks.clear',
+                        default: 'clear',
+                        description: 'clear canvas'
+                    })
+                }
+            ]
+        };
+    }
+
+    /**
+     * The pen "clear" block clears the pen layer's contents.
+     */
+    clear () {
+        const penSkinId = this._getPenLayerID();
+        if (penSkinId >= 0) {
+            this.runtime.renderer.penClear(penSkinId);
+            this.runtime.requestRedraw();
+        }
+
+        if (!this._posEmpty){
+            this._xPos.length = 0;
+            this._yPos.length = 0;
+            this._posEmpty = true;
+        }
+
+    }
+
+    /**
+     * The pen "pen down" block causes the target to leave pen trails on future motion.
+     * @param {object} args - the block arguments.
+     * @param {object} util - utility object provided by the runtime.
+     */
+    drawAxis (args, util) {
+        const target = util.target;
+        
+        // console.log('draw axis');
+        target.setVisible(false);
+
+        const penSkinId = this._getPenLayerID();
+        if (penSkinId >= 0) {
+            const penState = this._getPenState(target);
+            target.x = this._xCenter + this._width;
+            target.y = this._yCenter;
+            this.runtime.renderer.penLine(penSkinId, penState.penAttributes, this._xCenter, this._yCenter, target.x, target.y);
+
+            for (let i = 0; i * this._interval < this._width; i++){
+                const interval = i * this._interval;
+                this.runtime.renderer.penLine(penSkinId, penState.penAttributes, this._xCenter + interval, this._yCenter, this._xCenter + interval, this._yCenter + 5);
+                this.drawText(this._xCenter + interval, this._yCenter - 10, i, target, false);
+            }
+            
+            // console.log(target.getCostumes());
+            target.x = this._xCenter;
+            target.y = this._yCenter + this._height;
+            this.runtime.renderer.penLine(penSkinId, penState.penAttributes, this._xCenter, this._yCenter, target.x, target.y);
+            for (let i = 1; i * this._interval < this._height; i++){
+                const interval = i * this._interval;
+                this.runtime.renderer.penLine(penSkinId, penState.penAttributes, this._xCenter, this._yCenter + interval, this._xCenter + 5, this._yCenter + interval);
+                this.drawText(this._xCenter - 10, this._yCenter + interval, i, target, true);
+            }
+            
+            // console.log(target.getCostumes().length);
+            for (let i = 0, j = 1; i * this._interval < this._width && j * this._interval < this._height; i++, j++){
```

Comment:
Why is this block of code needed when the two for loops above already drew the text?

Reply:
That is a test code, now it's deleted.